test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/AssociatedMetadataProviderTest.cs

```diff
@@ -233,6 +233,42 @@ public void GetMetadataForProperties_SetsContainerAsExpected()
             Assert.Same(model, metadata[2].Container);
         }
 
+        [Fact]
+        public void GetMetadataForPropertiesFromModelMetadataType()
```

Comment:
should also test model and model metadata class hierarchies with following in both:
- properties defined only in the derived class
- properties derived only in the base class
- non-`virtual` properties overridden using `new`
- `virtual` properties overridden using `override`

probably best to do this in a new `ModelAttributesTest` class.  then this test method confirms core scenario works more end-to-end.